### DISCUSSION TITLE
Grafana Infinity Source Time Range ?Bug?

### DISCUSSION THREAD

**Discussion by Molteee** [grafana/grafana-infinity-datasource#610] [Q&A]
Hey there,

I'm using a Infinity Data Source in Grafana for an JSON web API.

When I'm setting the time range in grafana to for example the last 2 or 7 Days the Table only displays the Entrys from the start Date of the time range. Never the Entrys for the whole weak or time range

Is that a known thing for Infinity? 

Thank you in advance!

**Comment by yesoreyeram** [comment]
Infinity doesn't know about your API or data structure. so it can't filter on client side. Instead what you need to do is to pass grafana time range variables to your API via query parameters or request body. Search for grafana global variables and you see variables such as  $__to and $__from

**Reply by Molteee** [reply]
Thank you already :) I do have query parameters in the URL options like: 
key                      value
CreatedFrom      ${__from:date}
CreatedTo           ${__to:date}

Do I need to change them?

When I set the range to "last 6 hours" it perfectly displays all values, but when I set it to 24 hours it only displays the values from the day before till 12 am but not further. Or when i say the last 7 days it displays the values from 7 days ago but not further.

**Reply by yesoreyeram** [reply]
Check what the corresponding API response looks like? Does the target api respect the time range passed?

**Reply by mykael22000** [reply]
I'd suspect the date/time format isn't acceptable to your api.  Use the QueryInspector to see what it is sending. 

You might want to see what will work with curl and then try and work out how to make grafana/infinity generate it...

I've ended up with:
```
  "FIRST_DAY" : ${__from:date:YYYY-MM-DD},
  "FIRST_TIME" : ${__from:date:HH}:${__from:date:mm}:${__from:date:ss},
  "LAST_DAY" : ${__to:date:YYYY-MM-DD},
  "LAST_TIME" : ${__to:date:HH}:${__to:date:mm}:${__to:date:ss},
```
 
...simply to get the date and time in a format my API host can process.  Note 'HH' to get 24 hour time rather than 'hh' which only gets you 12 hour time (that took a while to track down ;-( ).
